pkg/api/types.go

```diff
@@ -2070,10 +2070,83 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
+// Type and constants for component type.
+type ComponentType string
+
+// ComponentType constants define the known types of components.
+// Each type may have multiple instances.
+// Each instance's name is prefixed by the type, with a random suffix. See SimpleNameGenerator.
```

Comment:
@karlkfi Pretty sure this is the call site you're looking for re: `GenerateName`: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/rest/create.go#L64